### PR TITLE
feat(mapping): refresher

### PR DESCRIPTION
**Pull request by hekike**
## Summary

Update in-memory table mapping periodically to pick up new tables.
I also renamed `Timer` to `ExecutionTimer` to avoid conflicts.

Implementation of: https://github.com/ClickHouse/clickhouse-kafka-connect/issues/174

## Checklist

Delete items not relevant to your PR:
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by hekike** [comment]
I'm not experienced with Java.  would appreciate a directional check on this PR before I spend more time writing tests.

File: src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java [hunk 1]
```diff
@@ -85,6 +88,7 @@ public ClickHouseSinkConfig(Map<String, String> props) {
         sslEnabled = Boolean.parseBoolean(props.getOrDefault(SSL_ENABLED,"false"));
         timeout = Integer.parseInt(props.getOrDefault(TIMEOUT_SECONDS, timeoutSecondsDefault.toString())) * MILLI_IN_A_SEC; // multiple in 1000 milli
         retry = Integer.parseInt(props.getOrDefault(RETRY_COUNT, retryCountDefault.toString()));
+        tableRefreshInterval = Long.parseLong(props.getOrDefault(TABLE_REFRESH_INTERVAL, tableRefreshIntervalDefault.toString())) * MILLI_IN_A_SEC; // multiple in 1000 milli
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java:91]: Do we have an option to disable the timer?

Comment (mzitnik) [reply src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java:91]: we need to update it in the documentation

File: src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java [hunk 2]
```diff
@@ -74,16 +74,30 @@ public boolean start(ClickHouseSinkConfig csc) {
...
+        if (tableList.isEmpty()) {
+            return;
+        }
+
+        // Adding new tables to mapping
+        // TODO: check Kafka Connect's topics name or topics regex config and
+        // only add tables to in-memory mapping that matches the topics we consume.
+        // TODO: remove dropped tables from mapping
+        for (Table table: tableList) {
             this.mapping.put(table.getName(), table);
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java:99]: This logic only adds new tables but does not remove the old ones (we are not clearing the old mapping potential memory leak)

**Review by mzitnik — CHANGES_REQUESTED** [review]

File: src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java (hunk 2 above)
Comment (hekike) [reply src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java:114]: Made mapping thread-safe and added code to remove dropped tables. Please take a look.

File: src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java (hunk 1 above)
Comment (mzitnik) [reply src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java:91]: can be found in that repo. https://github.com/ClickHouse/clickhouse-docs

File: src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java (hunk 2 above)
Comment (mzitnik) [reply src/main/java/com/clickhouse/kafka/connect/sink/db/ClickHouseWriter.java:114]: @hekike I would take another direction here keep this.mapping = new HashMap<>();
In the updateMapping create a new mapping and only in the end of the method switch between the references 
Read https://stackoverflow.com/questions/44283378/why-is-reference-assignment-atomic-in-java

**Review by mzitnik — CHANGES_REQUESTED** [review]

**Comment by mzitnik** [comment]
@hekike Can you also provide tests for the new feature

**Comment by hekike** [comment]
@mzitnik, thanks for the input. I refactored the update method to replace mapping by reference.
I also added a test where I create the table after starting, but I need to learn how to run the test locally, and CI doesn't run on PR. 

Can you please give me some guidance on how to run tests?

**Comment by mzitnik** [comment]
@hekike that should do the trick ```./gradlew test```
You are correct. We need to add [pull_request](https://github.com/ClickHouse/clickhouse-kafka-connect/blob/main/.github/workflows/tests.yaml#L3)

**Comment by hekike** [comment]
> @hekike that should do the trick `./gradlew test` You are correct. We need to add [pull_request](https://github.com/ClickHouse/clickhouse-kafka-connect/blob/main/.github/workflows/tests.yaml#L3)

Thanks. @mzitnik, I added an integration test but `./gradlew integrationTest` returns:
> Starting a Gradle Daemon, 1 incompatible Daemon could not be reused

It's unfortunately not added to GitHub Action, so I can't run it on the CI. I tried to [add it](https://github.com/ClickHouse/clickhouse-kafka-connect/pull/175/commits/34e4220c0b07b8025158c0be2a89486978d8a987), but it skips the `integrationTest`:
https://github.com/openmeterio/clickhouse-kafka-connect/actions/runs/6136004746/job/16650364167?pr=1

**Comment by mzitnik** [comment]
@hekike We will take care of ```integrationTest``` currently we are not running it as gitaction in the cloud since it takes too long please remove the ```integrationTest``` so we can merge your PR

**Comment by hekike** [comment]
@mzitnik, thank you. Appriciate the more hands on help here. I removed my integration test and CI changes from the PR.

**Comment by mzitnik** [comment]
@hekike 
One last thing: Please add this new feature to https://github.com/ClickHouse/clickhouse-kafka-connect/blob/main/CHANGELOG.md
Open a new version 1.0.1 in CHANGELOG.md and add the new feature

**Comment by hekike** [comment]
@mzitnik added to `CHANGELOG.md`.

**Comment by mzitnik** [comment]
@hekike you can contact me through our community slack

**Pull request merged**